js/common/modules/@arangodb/foxx/check-args.js

```diff
@@ -283,7 +344,47 @@ exports.validateModel = function (value) {
       not ${typeof model.fromClient}.
     `), warnings};
   }
-  return {value: {model, multiple}, error: null};
+
+  if (multiple) {
```

Comment:
We probably want to deprecate the "wrap a model or schema in an array to indicate it should match an array of this type" stuff before ArangoDB 4 to be able to cut this. This logic previously existed in various other places and has been consolidated here to allow us to provide a consistent `validate` method regardless of what the model looks like.